subsys/bluetooth/mesh/blob_cli.c

```diff
@@ -626,48 +626,43 @@ static void send_end(int err, void *user_data)
 
 static void info_get_tx(struct bt_mesh_blob_cli *cli, uint16_t dst)
 {
-	BT_MESH_MODEL_BUF_DEFINE(buf, BT_MESH_BLOB_OP_INFO_GET, 0);
-	bt_mesh_model_msg_init(&buf, BT_MESH_BLOB_OP_INFO_GET);
+	BT_MESH_MODEL_BUF_INIT(buf, BT_MESH_BLOB_OP_INFO_GET, 0);
 
 	tx(cli, dst, &buf);
 }
 
 static void xfer_start_tx(struct bt_mesh_blob_cli *cli, uint16_t dst)
 {
-	BT_MESH_MODEL_BUF_DEFINE(buf, BT_MESH_BLOB_OP_XFER_START, 16);
-	bt_mesh_model_msg_init(&buf, BT_MESH_BLOB_OP_XFER_START);
-	net_buf_simple_add_u8(&buf, cli->xfer->mode << 6);
-	net_buf_simple_add_le64(&buf, cli->xfer->id);
-	net_buf_simple_add_le32(&buf, cli->xfer->size);
-	net_buf_simple_add_u8(&buf, cli->xfer->block_size_log);
-	net_buf_simple_add_le16(&buf, BT_MESH_TX_SDU_MAX);
+	BT_MESH_MODEL_BUF_INIT(buf, BT_MESH_BLOB_OP_XFER_START, 16,
+			       cli->xfer->mode << 6,
+			       BT_BYTES_LIST_LE64(cli->xfer->id),
+			       BT_BYTES_LIST_LE32(cli->xfer->size),
+			       cli->xfer->block_size_log,
+			       BT_BYTES_LIST_LE16(BT_MESH_TX_SDU_MAX));
```

Comment:
Hmm, is this really more readable than it was before with `_add_<endiannes><size>`? For me the original version is more readable and understandable. And if code size also increased, why do we need this change?

Reply:
What do you mean code size also increased, after last commit, the code size reduced, that mean initialization to definition instead of functions will save code size.

Reply:
I don't find the current approach difficult to understand, but I find it very simple.

Reply:
From what I see [here](https://github.com/zephyrproject-rtos/zephyr/pull/67662#issuecomment-1903496846) the code size is slightly increased.

Reply:
The reason for the slightly larger code size is the addition of array initialization not initialization msg to definition